### PR TITLE
Surveys: Support PS formatting and Global Banned Words check

### PR DESCRIPTION
**Pull request by DeathlyPlays**
Tested and works

### CODE REVIEW

**Comment by Mystifi** [comment]
Wouldn't this make it possible for users to hyperlink shock sites under seemingly innocent text via <a> tags?

**Comment by Mystifi** [comment]
Is this necessary? This can get a little out-of-hand with some specific users on the server that I could (but will not) name. It's not really bad, but I am not sure if it's really necessary.

**Comment by DeathlyPlays** [comment]
Like.... Its been requested to be done :/

**Comment by Mystifi** [comment]
I would run Config.chatfilter over the input to make sure nothing inappropriate can be injected into the survey. Otherwise, it looks fine, and I'll merge it.

I would probably place this Config.chatfilter check within the validateAnswer function, and probably before the checking of room banned words. I would assume that global banword checking should take higher precedence.

**Comment by DeathlyPlays** [comment]
Hmm

**Comment by DeathlyPlays** [comment]
Like this @Mystifi ?

**Comment by ghost** [comment]
i isn’t defined



Also should have mod chat for surveys or nah?

File: spacialgaze-plugins/survey.js [hunk 1]
```diff
@@ -144,8 +144,16 @@ class Survey {
...
 
-function validateAnswer(room, message) {
+function validateAnswer(room, message, connection, user) {
+	let answer = this.repliers[i];
 	if (!room) return true;
+	if (Config.chatfilter) {
+		let filterResult = Config.chatfilter.call(this, answer, user, null, connection);
+		if (!filterResult) return;
+		if (answer !== filterResult) {
+			return this.errorReply("Your answer contains a globally banned word.");
```
Comment (HoeenCoder) [inline spacialgaze-plugins/survey.js]: I dont think `this` refers to the command context here.

File: spacialgaze-plugins/survey.js [hunk 2]
```diff
@@ -144,8 +144,16 @@ class Survey {
 	}
 }
 
-function validateAnswer(room, message) {
+function validateAnswer(room, message, connection, user) {
```
Comment (HoeenCoder) [inline spacialgaze-plugins/survey.js]: This function returns `true` if the message is allowed, and `false` if not.

File: spacialgaze-plugins/survey.js (hunk 1 above)
Comment (DeathlyPlays) [reply spacialgaze-plugins/survey.js]: My bad I did this all in a rush I did test the formatting portion not this validation for globally banned words

**Comment by DeathlyPlays** [comment]
*passes baton to Bladicon*

**Comment by ghost** [comment]
tbh once i get it right i was just going to give you the code kek

**Comment by DeathlyPlays** [comment]
Lol

**Comment by DeathlyPlays** [comment]
Up to you

**Pull request closed without merging**